src/uu/ls/src/ls.rs

```diff
@@ -290,6 +295,40 @@ enum IndicatorStyle {
     Classify,
 }
 
+struct HyperlinkData {
+    prefix: String,
+    url_escaper: UrlEscaper,
+    current_dir: OnceCell<Option<PathBuf>>,
+}
+
+impl HyperlinkData {
+    fn from_env() -> Self {
+        const PREFIX: &str = "\x1B]8;;file://";
+
+        let hostname_url_escaper = UrlEscaper::new();
+        let prefix = [
+            PREFIX,
+            hostname_url_escaper
+                .escape_os(&hostname::get().unwrap_or_default())
+                .unwrap_or_default()
+                .as_str(),
+        ]
+        .concat();
+        let url_escaper = hostname_url_escaper.into_path_escaper();
```

Comment:
What is the reason that this can't be done in new? Something to do with the path separator?